modules/database/src/main/java/com/intuit/wasabi/database/impl/DBITransaction.java

```diff
@@ -178,78 +197,82 @@ private Object perform(HandleBlock block) {
 
             throw ex;
         } finally {
-            if (!inTransaction) {
-                close();
+            // Close the handle (internally this call releases the JDBC connection to the pool)
+            if (handle!=null) {
```

Comment:
Let us just move this null check in the close() method and call close(handle) everywhere.